# HG changeset patch
# User Luca Greco <lgreco@mozilla.com>
# Date 1457098203 -3600
#      Fri Mar 04 14:30:03 2016 +0100
# Node ID 6bb23ae93417ba8be7cf97b385e896f463aa2f4b
# Parent  d53f6871ec137d77e57f53fe8cf90da8afe0e671
TEST: try and catch exceptions on Troubleshoot.snapshot

MozReview-Commit-ID: IDla3FJiP5F

diff --git a/gfx/tests/browser/browser_windowless_troubleshoot_crash.js b/gfx/tests/browser/browser_windowless_troubleshoot_crash.js
--- a/gfx/tests/browser/browser_windowless_troubleshoot_crash.js
+++ b/gfx/tests/browser/browser_windowless_troubleshoot_crash.js
@@ -28,18 +28,28 @@ add_task(function* test_windowlessBrowse
   let winUtils = webNav.document.defaultView.
                         QueryInterface(Ci.nsIInterfaceRequestor).
                         getInterface(Ci.nsIDOMWindowUtils);
   is(winUtils.layerManagerType, "None", "windowless browser's layerManagerType should be 'None'");
 
   ok(true, "not crashed");
 
   var Troubleshoot = Cu.import("resource://gre/modules/Troubleshoot.jsm", {}).Troubleshoot;
-  var data = yield new Promise((resolve, reject) => {
-    Troubleshoot.snapshot((data) => {
-      resolve(data);
+
+  try {
+    var data = yield new Promise((resolve, reject) => {
+      try {
+        Troubleshoot.snapshot((data) => {
+          resolve(data);
+        });
+      } catch(e) {
+        reject(e);
+      }
     });
-  });
 
-  ok(data.graphics.windowLayerManagerType !== "None", "windowless browser window should not set windowLayerManagerType to 'None'");
+    ok(data.graphics.windowLayerManagerType !== "None", "windowless browser window should not set windowLayerManagerType to 'None'");
+
+  } catch(e) {
+    is(e, null, "unexpected exception on Troubleshoot.snapshot");
+  }
 
   webNav.close();
 });
